lib/nexmo_markdown_renderer/filters/snippet_variables_filter.rb

```diff
@@ -0,0 +1,33 @@
+module Nexmo
+  module Markdown
+    class SnippetVariablesFilter < Banzai::Filter
+      def call(input)
+        input.gsub(/```snippet_variables(.+?)```/m) do |_s|
+          config = YAML.safe_load(Regexp.last_match(1))
+
+          raise 'No variables provided' unless config
+          raise 'Must provide a list' unless config.is_a?(Array)
+
+          output = <<~HEREDOC
+            Key | Description
+            -- | --
+          HEREDOC
+          config.each do |key|
+            details = variables[key]
+            raise "#{key} is not a valid snippet variable" unless details
+
+            output += <<~HEREDOC
+              `#{key}` | #{details['description']}
+            HEREDOC
+          end
+
+          output
+        end
+      end
+
+      def variables
+        @variables ||= YAML.safe_load(File.read("#{Nexmo::Markdown::Config.docs_base_path}/config/code_snippet_variables.yml"))
```

Comment:
How about adding a `raise` if the config file does not exist? Could help guide the correct setup here.

Reply:
The standard error gives us enough information here

```
Errno::ENOENT: No such file or directory @ rb_sysopen - spec/fixtures/config/code_snippet_variables.yml
```

Reply:
Makes sense